crates/bevy_render/src/camera/camera.rs

```diff
@@ -138,6 +142,80 @@ impl Camera {
             None
         }
     }
+
+    /// Given a position in screen space, compute the world-space line that corresponds to it.
+    pub fn screen_to_world_ray(
+        &self,
+        pos_screen: Vec2,
+        windows: &Windows,
+        images: &Assets<Image>,
+        camera_transform: &GlobalTransform,
+    ) -> Line {
+        let view_matrix = camera_transform.compute_matrix();
+        let window_size = self.target.get_logical_size(windows, images).unwrap();
+        let projection_matrix = self.projection_matrix;
+
+        // Normalized device coordinate cursor position from (-1, -1, 1) to (1, 1, 0) where 0 is at the far plane
+        // and 1 is at the near plane.
+        let cursor_ndc = (pos_screen / window_size) * 2.0 - Vec2::ONE;
+        let cursor_pos_ndc_near: Vec3 = cursor_ndc.extend(1.0);
+        let cursor_pos_ndc_far: Vec3 = cursor_ndc.extend(0.0);
+
+        // Use near and far ndc points to generate a ray in world space
+        // This method is more robust than using the location of the camera as the start of
+        // the ray, because ortho cameras have a focal point at infinity!
+        let inverse_projection = projection_matrix.inverse();
+        let cursor_pos_view_near = inverse_projection.project_point3(cursor_pos_ndc_near);
+        let cursor_pos_view_far = inverse_projection.project_point3(cursor_pos_ndc_far);
```

Comment:
Isn't this projecting to infinity, since the far plane is at -Infinity Z for reversed-Z perspective projection?

Reply:
I just noticed this is [taken directly from an old version of `bevy_mod_raycast`](https://github.com/aevyrie/bevy_mod_raycast/blob/82cb6824983085509b58723ada1846826de5b1f4/src/primitives.rs#L131-L133), which does not work with the new projections for the reason you stated.

Here is the current version of that function that works with bevy: https://github.com/aevyrie/bevy_mod_raycast/blob/main/src/primitives.rs#L180-L217

Reply:
Also, the code "taken" is from a repo with an MIT license. I could re-license that repo as MIT/Apache if I get permission from contributors, but I'm not very familiar with licensing laws. The math here is well understood and fairly trivial, the only issue I can see is there was an obvious copy-paste.

Reply:
A simpler solution is to use z = 1.0 for the near plane and z = 0.5 for half-way to the far plane (just kidding, it works though.)